In [1]:
import pandas as pd
import re

import numpy as np

In [2]:
annual_d_r =  pd.read_csv('datasets/annual-number-of-deaths-by-cause.csv')

In [3]:
annual_d_r.head()

Entity Code  Year Number of executions (Amnesty International)  \
0  Afghanistan  AFG  2007                                           15   
1  Afghanistan  AFG  2008                                           17   
2  Afghanistan  AFG  2009                                            0   
3  Afghanistan  AFG  2011                                            2   
4  Afghanistan  AFG  2012                                           14   

   Deaths - Meningitis - Sex: Both - Age: All Ages (Number)  \
0                                        2932.558591          
1                                        2730.846137          
2                                        2460.186423          
3                                        2326.767573          
4                                        2254.492968          

   Deaths - Neoplasms - Sex: Both - Age: All Ages (Number)  \
0                                       15924.782541         
1                                       16147.839026         
2                                       16382.558546         
3                                       17093.515668         
4                                       17521.930189         

   Deaths - Fire, heat, and hot substances - Sex: Both - Age: All Ages (Number)  \
0                                         481.219817                              
1                                         462.180576                              
2                                         448.300264                              
3                                         447.884354                              
4                                         445.226412                              

   Deaths - Malaria - Sex: Both - Age: All Ages (Number)  \
0                                         392.942607       
1                                         254.703403       
2                                         239.035635       
3                                         390.060248       
4                                          93.894201       

   Deaths - Drowning - Sex: Both - Age: All Ages (Number)  \
0                                        2126.788535        
1                                        1973.375893        
2                                        1852.155239        
3                                        1774.968476        
4                                        1715.864705        

   Deaths - Interpersonal violence - Sex: Both - Age: All Ages (Number)  ...  \
0                                        3657.205757                     ...   
1                                        3784.702895                     ...   
2                                        3874.099828                     ...   
3                                        4169.873038                     ...   
4                                        4244.671108                     ...   

   Deaths - Protein-energy malnutrition - Sex: Both - Age: All Ages (Number)  \
0                                        2438.567035                           
1                                        2230.538625                           
2                                        1997.531216                           
3                                        1805.467089                           
4                                        1666.898963                           

   Terrorism (deaths)  \
0              1199.0   
1              1092.0   
2              1065.0   
3              1525.0   
4              3521.0   

   Deaths - Cardiovascular diseases - Sex: Both - Age: All Ages (Number)  \
0                                       53962.251859                       
1                                       54050.644196                       
2                                       53964.195557                       
3                                       54347.266593                       
4                                       54868.120368                       

   Deaths - Ch

In [4]:
#rename columns name
annual_d_r.rename(columns={'Entity':'country', 'Code':'code', 'Year':'year',
       'Number of executions (Amnesty International)':'number_of_executions',
       'Deaths - Meningitis - Sex: Both - Age: All Ages (Number)':'meningitis',
       'Deaths - Neoplasms - Sex: Both - Age: All Ages (Number)':'neoplasms',
       'Deaths - Fire, heat, and hot substances - Sex: Both - Age: All Ages (Number)':'fire_heat_hot_substances',
       'Deaths - Malaria - Sex: Both - Age: All Ages (Number)':'malaria',
       'Deaths - Drowning - Sex: Both - Age: All Ages (Number)':'drowing',
       'Deaths - Interpersonal violence - Sex: Both - Age: All Ages (Number)':'interpersonal_violence',
       'Deaths - HIV/AIDS - Sex: Both - Age: All Ages (Number)':'hiv_aids',
       'Deaths - Drug use disorders - Sex: Both - Age: All Ages (Number)':'drug_use_disorders',
       'Deaths - Tuberculosis - Sex: Both - Age: All Ages (Number)':'tuberculosis',
       'Deaths - Road injuries - Sex: Both - Age: All Ages (Number)':'road_injuries',
       'Deaths - Maternal disorders - Sex: Both - Age: All Ages (Number)':'maternal_disorders',
       'Deaths - Lower respiratory infections - Sex: Both - Age: All Ages (Number)':'lower_respiratory_infection',
       'Deaths - Neonatal disorders - Sex: Both - Age: All Ages (Number)':'neonatal_disorder',
       'Deaths - Alcohol use disorders - Sex: Both - Age: All Ages (Number)':'alcohol_use_disorders',
       'Deaths - Exposure to forces of nature - Sex: Both - Age: All Ages (Number)':'force_of_nature',
       'Deaths - Diarrheal diseases - Sex: Both - Age: All Ages (Number)':'diarrheal',
       'Deaths - Environmental heat and cold exposure - Sex: Both - Age: All Ages (Number)':'environmental_heat_cold',
       'Deaths - Nutritional deficiencies - Sex: Both - Age: All Ages (Number)':'nutritional_deficiencies',
       'Deaths - Self-harm - Sex: Both - Age: All Ages (Number)':'self_harm',
       'Deaths - Conflict and terrorism - Sex: Both - Age: All Ages (Number)':'conflict_and_terrorism',
       'Deaths - Diabetes mellitus - Sex: Both - Age: All Ages (Number)':'diabetes_mellitus',
       'Deaths - Poisonings - Sex: Both - Age: All Ages (Number)':'poisoning',
       'Deaths - Protein-energy malnutrition - Sex: Both - Age: All Ages (Number)':'protein_energy_malnutrition',
       'Terrorism (deaths)':'terrorism',
       'Deaths - Cardiovascular diseases - Sex: Both - Age: All Ages (Number)':'cardiovascular_diseases',
       'Deaths - Chronic kidney disease - Sex: Both - Age: All Ages (Number)':'chronic_kidney_disease',
       'Deaths - Chronic respiratory diseases - Sex: Both - Age: All Ages (Number)':'chronic_respiratory_deasease',
       'Deaths - Cirrhosis and other chronic liver diseases - Sex: Both - Age: All Ages (Number)':'cirrohosis_liver_deasease',
       'Deaths - Digestive diseases - Sex: Both - Age: All Ages (Number)':'digestive_disease',
       'Deaths - Acute hepatitis - Sex: Both - Age: All Ages (Number)':'acute_hepatitis',"Deaths - Alzheimer's disease and other dementias - Sex: Both - Age: All Ages (Number)":"alzhimers_disease",
       "Deaths - Parkinson's disease - Sex: Both - Age: All Ages (Number)":"parkinsons_disease"
       }, inplace = True)

In [5]:
annual_d_r.head()

country code  year number_of_executions   meningitis     neoplasms  \
0  Afghanistan  AFG  2007                   15  2932.558591  15924.782541   
1  Afghanistan  AFG  2008                   17  2730.846137  16147.839026   
2  Afghanistan  AFG  2009                    0  2460.186423  16382.558546   
3  Afghanistan  AFG  2011                    2  2326.767573  17093.515668   
4  Afghanistan  AFG  2012                   14  2254.492968  17521.930189   

   fire_heat_hot_substances     malaria      drowing  interpersonal_violence  \
0                481.219817  392.942607  2126.788535             3657.205757   
1                462.180576  254.703403  1973.375893             3784.702895   
2                448.300264  239.035635  1852.155239             3874.099828   
3                447.884354  390.060248  1774.968476             4169.873038   
4                445.226412   93.894201  1715.864705             4244.671108   

   ...  protein_energy_malnutrition  terrorism  cardiovascular_diseases  \
0  ...                  2438.567035     1199.0             53962.251859   
1  ...                  2230.538625     1092.0             54050.644196   
2  ...                  1997.531216     1065.0             53964.195557   
3  ...                  1805.467089     1525.0             54347.266593   
4  ...                  1666.898963     3521.0             54868.120368   

   chronic_kidney_disease  chronic_respiratory_deasease  \
0             4490.174240                   7221.773068   
1             4533.825139                   7142.796656   
2             4596.974422                   7045.249090   
3             4784.629776                   6915.563606   
4             4845.667028                   6877.511217   

   cirrohosis_liver_deasease  digestive_disease  acute_hepatitis  \
0                3345.814979        6457.753476           3437.0   
1                3316.354776        6407.964590           3005.0   
2                3291.158847        6358.027600           2663.0   
3                3317.891662        6370.000383           2365.0   
4                3352.846566        6398.171770           2264.0   

   alzhimers_disease  parkinsons_disease  
0             1402.0               450.0  
1             1424.0               455.0  
2             1449.0               460.0  
3             1508.0               473.0  
4             1544.0               482.0  

[5 rows x 36 columns]

In [100]:
annual_d_r.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8590 entries, 0 to 8589
Data columns (total 36 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   country                       8590 non-null   object 
 1   code                          6206 non-null   object 
 2   year                          8590 non-null   int64  
 3   number_of_executions          267 non-null    object 
 4   meningitis                    8220 non-null   float64
 5   neoplasms                     8220 non-null   float64
 6   fire_heat_hot_substances      8220 non-null   float64
 7   malaria                       8220 non-null   float64
 8   drowing                       8220 non-null   float64
 9   interpersonal_violence        8220 non-null   float64
 10  hiv_aids                      8220 non-null   float64
 11  drug_use_disorders            8220 non-null   float64
 12  tuberculosis                  8220 non-null   float64
 13  roa

#saving the renamed _data


annual_d_r.to_csv('datasets/renamed_data.csv')

In [6]:
#loading population data
total_p = pd.read_csv('datasets/WPP2019_TotalPopulationBySex.csv')
total_p.head()

LocID     Location  VarID Variant  Time  MidPeriod   PopMale  PopFemale  \
0      4  Afghanistan      2  Medium  1950     1950.5  4099.243   3652.874   
1      4  Afghanistan      2  Medium  1951     1951.5  4134.756   3705.395   
2      4  Afghanistan      2  Medium  1952     1952.5  4174.450   3761.546   
3      4  Afghanistan      2  Medium  1953     1953.5  4218.336   3821.348   
4      4  Afghanistan      2  Medium  1954     1954.5  4266.484   3884.832   

   PopTotal  PopDensity  
0  7752.117      11.874  
1  7840.151      12.009  
2  7935.996      12.156  
3  8039.684      12.315  
4  8151.316      12.486

In [7]:
#filtering data from 1990 to 2019
total_sub = total_p[(total_p['Time']>1989) & (total_p['Time']<2020)]

In [8]:
#loading country data
country = pd.read_excel('datasets/countries name.xlsx')

In [9]:
#showing random 5 data from country datasets
country.sample(5)

country   latitude  longitude      name
105      IR  32.427908  53.688046      Iran
74       GD  12.262776 -61.604171   Grenada
163      NO  60.472024   8.468946    Norway
174      PK  30.375321  69.345116  Pakistan
136      MD  47.411631  28.369885   Moldova

In [10]:
# filtering by country 
population_all_countries= total_sub[total_sub['Location'].isin(country.name.unique().tolist())]

In [11]:
#dropping column latitude and longitude
population_all_countries = population_all_countries[['Location', 'PopTotal', 'Time']]

In [12]:
# first 5 data
population_all_countries.head()

Location   PopTotal  Time
40  Afghanistan  12412.311  1990
41  Afghanistan  13299.016  1991
42  Afghanistan  14485.543  1992
43  Afghanistan  15816.601  1993
44  Afghanistan  17075.728  1994

In [16]:
#merging the data to main data by matching country name
annual_t_d__r = annual_d_r[annual_d_r['country'].isin(population_all_countries.Location.unique().tolist())]

In [17]:
#these countries didnt match the list
country[~country['name'].isin(annual_d_r.country.unique().tolist())]['name']

4                                          Anguilla
7                              Netherlands Antilles
9                                        Antarctica
14                                            Aruba
31                                    Bouvet Island
36                          Cocos [Keeling] Islands
39                                 Congo [Republic]
41                                    Côte d'Ivoire
50                                 Christmas Island
52                                   Czech Republic
68                Falkland Islands [Islas Malvinas]
69                                       Micronesia
70                                    Faroe Islands
77                                         Guernsey
79                                        Gibraltar
86     South Georgia and the South Sandwich Islands
91                                       Gaza Strip
93                Heard Island and McDonald Islands
101                                     Isle of Man
103         

In [18]:
annual_d_r[ ~annual_d_r['country'].isin(population_all_countries.Location.unique().tolist())].country.unique()

array(['Africa', 'African Region', 'African Union', 'America',
       'Andean Latin America', 'Asia', 'Australasia',
       'Australasia & Oceania', 'Bolivia', 'Bosnia-Herzegovina', 'Brunei',
       'Cape Verde', 'Caribbean', 'Central America & Caribbean',
       'Central Asia', 'Central Europe',
       'Central Europe, Eastern Europe, and Central Asia',
       'Central Latin America', 'Central sub-Saharan Africa',
       'Commonwealth', 'Commonwealth High Income',
       'Commonwealth Low Income', 'Commonwealth Middle Income',
       "Cote d'Ivoire", 'Czechia', 'Czechoslovakia',
       'Democratic Republic of Congo', 'East Asia', 'East Asia & Pacific',
       'East Asia & Pacific - WB', 'East Germany (GDR)', 'East Timor',
       'Eastern Europe', 'Eastern Mediterranean Region',
       'Eastern sub-Saharan Africa', 'England', 'Eswatini', 'Europe',
       'Europe & Central Asia', 'Europe & Central Asia - WB',
       'European Region', 'European Union', 'Four World Regions', 'G20',
     

In [19]:
#rename the columns name
population_all_countries.rename(columns = {'Time':'year', 'Location':'country'}, inplace = True )

In [20]:
# merging data by country and year
df = pd.merge(annual_t_d__r, population_all_countries, on = ['country', 'year'])

In [21]:

df.country.nunique()

188

we have 188 countries data of death by desease

In [22]:
df.shape

(5469, 37)

There are 5469 rows and 37 columns of data

In [23]:
#loading data of gdp
gdp = pd.read_csv('datasets/gdp-per-capita-worldbank.csv')

In [24]:
gdp.drop(columns = 'Code', inplace = True)

In [25]:
gdp.columns = ['country', 'year', 'gdp-per-capita']

In [26]:
gdp.head()

country  year  gdp-per-capita
0  Afghanistan  2002     1189.784668
1  Afghanistan  2003     1235.810063
2  Afghanistan  2004     1200.278013
3  Afghanistan  2005     1286.793659
4  Afghanistan  2006     1315.789117

In [27]:
# merging the gdp data with main
df_gdp = pd.merge(df, gdp,how = 'left',  on = ['country','year'])

In [28]:
df_gdp.shape

(5469, 38)

In [29]:
df_gdp.isnull().sum()

country                            0
code                               0
year                               0
number_of_executions            5253
meningitis                         9
neoplasms                          9
fire_heat_hot_substances           9
malaria                            9
drowing                            9
interpersonal_violence             9
hiv_aids                           9
drug_use_disorders                 9
tuberculosis                       9
road_injuries                      9
maternal_disorders                 9
lower_respiratory_infection        9
neonatal_disorder                  9
alcohol_use_disorders              9
force_of_nature                    9
diarrheal                          9
environmental_heat_cold            9
nutritional_deficiencies           9
self_harm                          9
conflict_and_terrorism             9
diabetes_mellitus                  9
poisoning                          9
protein_energy_malnutrition        9
t

In [30]:
df_gdp.to_csv('all_combined_annual_death_by_cause.csv', index= False)

In [31]:
df = pd.read_csv('all_combined_annual_death_by_cause.csv')

In [32]:
df.isnull().sum()/len(df)*100

country                          0.000000
code                             0.000000
year                             0.000000
number_of_executions            96.050466
meningitis                       0.164564
neoplasms                        0.164564
fire_heat_hot_substances         0.164564
malaria                          0.164564
drowing                          0.164564
interpersonal_violence           0.164564
hiv_aids                         0.164564
drug_use_disorders               0.164564
tuberculosis                     0.164564
road_injuries                    0.164564
maternal_disorders               0.164564
lower_respiratory_infection      0.164564
neonatal_disorder                0.164564
alcohol_use_disorders            0.164564
force_of_nature                  0.164564
diarrheal                        0.164564
environmental_heat_cold          0.164564
nutritional_deficiencies         0.164564
self_harm                        0.164564
conflict_and_terrorism           0

10 percent of data on gdp were missing

In [33]:
#continent data
cont = pd.read_excel('Book2.xlsx')



#renaming continent columns
cont.rename(columns = {'Country_Number':'code', 'continent_name':'continent'}, inplace = True)

#mergind continend data with main data
df_c = pd.merge(df,cont, how = 'left',on = ['code'])

Index(['country', 'code', 'year', 'number_of_executions', 'meningitis',
       'neoplasms', 'fire_heat_hot_substances', 'malaria', 'drowing',
       'interpersonal_violence', 'hiv_aids', 'drug_use_disorders',
       'tuberculosis', 'road_injuries', 'maternal_disorders',
       'lower_respiratory_infection', 'neonatal_disorder',
       'alcohol_use_disorders', 'force_of_nature', 'diarrheal',
       'environmental_heat_cold', 'nutritional_deficiencies', 'self_harm',
       'conflict_and_terrorism', 'diabetes_mellitus', 'poisoning',
       'protein_energy_malnutrition', 'terrorism', 'cardiovascular_diseases',
       'chronic_kidney_disease', 'chronic_respiratory_deasease',
       'cirrohosis_liver_deasease', 'digestive_disease', 'acute_hepatitis',
       'alzhimers_disease', 'parkinsons_disease', 'PopTotal', 'gdp-per-capita',
       'continent'],
      dtype='object')

In [34]:
df_c.head()

country code  year number_of_executions   meningitis     neoplasms  \
0  Afghanistan  AFG  2007                   15  2932.558591  15924.782541   
1  Afghanistan  AFG  2008                   17  2730.846137  16147.839026   
2  Afghanistan  AFG  2009                    0  2460.186423  16382.558546   
3  Afghanistan  AFG  2011                    2  2326.767573  17093.515668   
4  Afghanistan  AFG  2012                   14  2254.492968  17521.930189   

   fire_heat_hot_substances     malaria      drowing  interpersonal_violence  \
0                481.219817  392.942607  2126.788535             3657.205757   
1                462.180576  254.703403  1973.375893             3784.702895   
2                448.300264  239.035635  1852.155239             3874.099828   
3                447.884354  390.060248  1774.968476             4169.873038   
4                445.226412   93.894201  1715.864705             4244.671108   

   ...  chronic_kidney_disease  chronic_respiratory_deasease  \
0  ...             4490.174240                   7221.773068   
1  ...             4533.825139                   7142.796656   
2  ...             4596.974422                   7045.249090   
3  ...             4784.629776                   6915.563606   
4  ...             4845.667028                   6877.511217   

   cirrohosis_liver_deasease  digestive_disease  acute_hepatitis  \
0                3345.814979        6457.753476           3437.0   
1                3316.354776        6407.964590           3005.0   
2                3291.158847        6358.027600           2663.0   
3                3317.891662        6370.000383           2365.0   
4                3352.846566        6398.171770           2264.0   

   alzhimers_disease  parkinsons_disease   PopTotal  gdp-per-capita  continent  
0             1402.0               450.0  27100.542     1460.825751       Asia  
1             1424.0               455.0  27722.281     1484.114461       Asia  
2             1449.0               460.0  28394.806     1758.904477       Asia  
3             1508.0               473.0  30117.411     1904.559926       Asia  
4             1544.0               482.0  31161.378     2075.491614       Asia  

[5 rows x 39 columns]

In [39]:
## Merging continent on the data

In [40]:
#saving as csv
df_c.to_csv('all_combined_annual_death_by_cause.csv', index= False)

In [42]:
all_c_data = pd.read_csv('all_combined_annual_death_by_cause.csv')

all_c_data.head()

In [44]:
#dropping columns of death that is not caused by disease
all_c_data.drop(columns =['number_of_executions','terrorism'], inplace=True)

In [47]:
#there were missing data in all of digestive disease so we remove the missing data
all_c_data = all_c_data[~all_c_data['digestive_disease'].isnull()]

In [49]:
all_c_data.to_csv('all_combined_annual_death_by_cause.csv', index= False)

In [85]:
#loading cancer data
cancer_type = pd.read_csv('datasets/total-cancer-deaths-by-type.csv')
cancer_type.head()

Entity Code  Year  \
0  Afghanistan  AFG  1990   
1  Afghanistan  AFG  1991   
2  Afghanistan  AFG  1992   
3  Afghanistan  AFG  1993   
4  Afghanistan  AFG  1994   

   Deaths - Liver cancer - Sex: Both - Age: All Ages (Number)  \
0                                                851            
1                                                866            
2                                                890            
3                                                914            
4                                                933            

   Deaths - Kidney cancer - Sex: Both - Age: All Ages (Number)  \
0                                                 66             
1                                                 66             
2                                                 68             
3                                                 70             
4                                                 71             

   Deaths - Lip and oral cavity cancer - Sex: Both - Age: All Ages (Number)  \
0                                                 89                          
1                                                 89                          
2                                                 91                          
3                                                 93                          
4                                                 94                          

   Deaths - Tracheal, bronchus, and lung cancer - Sex: Both - Age: All Ages (Number)  \
0                                                983                                   
1                                                982                                   
2                                                989                                   
3                                                995                                   
4                                                996                                   

   Deaths - Larynx cancer - Sex: Both - Age: All Ages (Number)  \
0                                                260             
1                                                263             
2                                                268             
3                                                275             
4                                                282             

   Deaths - Gallbladder and biliary tract cancer - Sex: Both - Age: All Ages (Number)  \
0                                                180                                    
1                                                182                                    
2                                                185                                    
3                                                189                                    
4                                                193                                    

   Deaths - Malignant skin melanoma - Sex: Both - Age: All Ages (Number)  ...  \
0                                                 47                      ...   
1                                                 48                      ...   
2                                                 51                      ...   
3                                                 53                      ...   
4                                                 54                      ...   

   Deaths - Brain and central nervous system cancer - Sex: Both - Age: All Ages (Number)  \
0                                                422                                       
1                                                438                                       
2                                                472                                       
3                                                505                                       
4                                                526                                       

   Deaths - Non-Hodgkin lymphoma - Sex: Both - Age: All Ages (Number)

In [86]:
#renaming columns name
cancer_type.rename(columns = 
                  {
                      'Entity':'country', 'Code':'code', 'Year':'year',
       'Deaths - Liver cancer - Sex: Both - Age: All Ages (Number)':'liver_cancer',
       'Deaths - Kidney cancer - Sex: Both - Age: All Ages (Number)':'kidney_cancer',
       'Deaths - Lip and oral cavity cancer - Sex: Both - Age: All Ages (Number)':'lip_and_oral_cavity_cancer',
       'Deaths - Tracheal, bronchus, and lung cancer - Sex: Both - Age: All Ages (Number)':'lung',
       'Deaths - Larynx cancer - Sex: Both - Age: All Ages (Number)':'larynx',
       'Deaths - Gallbladder and biliary tract cancer - Sex: Both - Age: All Ages (Number)':'gallbladder_biliary',
       'Deaths - Malignant skin melanoma - Sex: Both - Age: All Ages (Number)':'malignant_skin',
       'Deaths - Leukemia - Sex: Both - Age: All Ages (Number)':'lukemia',
       'Deaths - Hodgkin lymphoma - Sex: Both - Age: All Ages (Number)':'hodgkin_lymphoma',
       'Deaths - Multiple myeloma - Sex: Both - Age: All Ages (Number)':'multiple_myeloma',
       'Deaths - Other neoplasms - Sex: Both - Age: All Ages (Number)':'other',
       'Deaths - Breast cancer - Sex: Both - Age: All Ages (Number)':'breast',
       'Deaths - Prostate cancer - Sex: Both - Age: All Ages (Number)':'prostate',
       'Deaths - Thyroid cancer - Sex: Both - Age: All Ages (Number)':'thyroid',
       'Deaths - Stomach cancer - Sex: Both - Age: All Ages (Number)':'stomach',
       'Deaths - Bladder cancer - Sex: Both - Age: All Ages (Number)':'bladder',
       'Deaths - Uterine cancer - Sex: Both - Age: All Ages (Number)':'uterine',
       'Deaths - Ovarian cancer - Sex: Both - Age: All Ages (Number)':'ovarian',
       'Deaths - Cervical cancer - Sex: Both - Age: All Ages (Number)':'cervical',
       'Deaths - Brain and central nervous system cancer - Sex: Both - Age: All Ages (Number)':'brain',
       'Deaths - Non-Hodgkin lymphoma - Sex: Both - Age: All Ages (Number)':'non_hodgkin_lymphoma',
       'Deaths - Pancreatic cancer - Sex: Both - Age: All Ages (Number)':'pancreatic',
       'Deaths - Esophageal cancer - Sex: Both - Age: All Ages (Number)':'esophageal',
       'Deaths - Testicular cancer - Sex: Both - Age: All Ages (Number)':'testicular',
       'Deaths - Nasopharynx cancer - Sex: Both - Age: All Ages (Number)':'nasopharynx',
       'Deaths - Other pharynx cancer - Sex: Both - Age: All Ages (Number)':'other_pharynx',
       'Deaths - Colon and rectum cancer - Sex: Both - Age: All Ages (Number)':'colon',
       'Deaths - Non-melanoma skin cancer - Sex: Both - Age: All Ages (Number)':'non_melanoma_skin',
       'Deaths - Mesothelioma - Sex: Both - Age: All Ages (Number)':'mesothelioma',
                      
                  }, inplace = True)

In [87]:
#dropping code columns
cancer_type.drop(columns = 'code', inplace = True)

#saving cancer data as csv
cancer_type.to_csv('cancer_type.csv', index = False)

In [88]:
df_c.columns

Index(['country', 'code', 'year', 'number_of_executions', 'meningitis',
       'neoplasms', 'fire_heat_hot_substances', 'malaria', 'drowing',
       'interpersonal_violence', 'hiv_aids', 'drug_use_disorders',
       'tuberculosis', 'road_injuries', 'maternal_disorders',
       'lower_respiratory_infection', 'neonatal_disorder',
       'alcohol_use_disorders', 'force_of_nature', 'diarrheal',
       'environmental_heat_cold', 'nutritional_deficiencies', 'self_harm',
       'conflict_and_terrorism', 'diabetes_mellitus', 'poisoning',
       'protein_energy_malnutrition', 'terrorism', 'cardiovascular_diseases',
       'chronic_kidney_disease', 'chronic_respiratory_deasease',
       'cirrohosis_liver_deasease', 'digestive_disease', 'acute_hepatitis',
       'alzhimers_disease', 'parkinsons_disease', 'PopTotal', 'gdp-per-capita',
       'continent'],
      dtype='object')

In [60]:
cc = ['meningitis',
       'neoplasms', 'fire_heat_hot_substances', 'malaria', 'drowing',
       'interpersonal_violence', 'hiv_aids', 'drug_use_disorders',
       'tuberculosis', 'road_injuries', 'maternal_disorders',
       'lower_respiratory_infection', 'neonatal_disorder',
       'alcohol_use_disorders', 'force_of_nature', 'diarrheal',
       'environmental_heat_cold', 'nutritional_deficiencies', 'self_harm',
       'conflict_and_terrorism', 'diabetes_mellitus', 'poisoning',
       'protein_energy_malnutrition', 'terrorism', 'cardiovascular_diseases',
       'chronic_kidney_disease', 'chronic_respiratory_deasease',
       'cirrohosis_liver_deasease', 'digestive_disease', 'acute_hepatitis',
       'alzhimers_disease', 'parkinsons_disease', 'gdp-per-capita']

meningitis     neoplasms  fire_heat_hot_substances      malaria  \
0     2932.558591  15924.782541                481.219817   392.942607   
1     2730.846137  16147.839026                462.180576   254.703403   
2     2460.186423  16382.558546                448.300264   239.035635   
3     2326.767573  17093.515668                447.884354   390.060248   
4     2254.492968  17521.930189                445.226412    93.894201   
...           ...           ...                       ...          ...   
5464  1438.676761  11161.386422                632.349622  2517.901527   
5465  1457.360801  11464.883288                648.435308  2050.099318   
5466  1459.529824  11743.672939                653.854499  2116.295646   
5467  1449.851578  12037.819441                657.215472  2088.361260   
5468  1450.049724  12352.632022                661.997695  2068.185673   

          drowing  interpersonal_violence      hiv_aids  drug_use_disorders  \
0     2126.788535             3657.205757    148.276329          251.830065   
1     1973.375893             3784.702895    157.401135          261.391617   
2     1852.155239             3874.099828    167.270695          269.845287   
3     1774.968476             4169.873038    183.816343          292.312638   
4     1715.864705             4244.671108    190.525008          304.570153   
...           ...                     ...           ...                 ...   
5464   769.587871             1302.008154  29161.730985          103.775228   
5465   801.263047             1342.122199  27140.790944          109.677478   
5466   818.357585             1363.089990  24846.237930          115.304607   
5467   824.663639             1395.971181  22106.464437          121.004527   
5468   827.056395             1433.595414  20722.240154          127.073539   

      tuberculosis  road_injuries  ...  terrorism  cardiovascular_diseases  \
0      4995.103369    7425.071926  ...     1199.0             53962.251859   
1      4789.722282    7354.749003  ...     1092.0             54050.644196   
2      4578.915137    7290.468621  ...     1065.0             53964.195557   
3      4258.825531    7432.300560  ...     1525.0             54347.266593   
4      4121.942028    7494.351481  ...     3521.0             54868.120368   
...            ...            ...  ...        ...                      ...   
5464  11213.954856    2372.593805  ...        NaN             16649.456872   
5465  10998.488251    2435.995792  ...        NaN             16937.071816   
5466  10762.490965    2473.357996  ...        0.0             17187.336508   
5467  10544.745280    2509.425290  ...        NaN             17460.197354   
5468  10465.151975    2553.770709  ...        NaN             17810.044243   

      chronic_kidney_disease  chronic_respiratory_deasease  \
0                4490.174240                   7221.773068   
1                4533.825139                   7142.796656   
2                4596.974422                   7045.249090   
3                4784.629776                   6915.563606   
4                4845.667028                   6877.511217   
...                      ...                           ...   
5464             2107.942765                   2750.515581   
5465             2159.865077                   2788.121829   
5466             2196.023891                   2818.465694   
5467             2240.368531                   2849.100567   
5468             2292.252230                   2891.114470   

      cirrohosis_liver_deasease  digestive_disease  acute_hepatitis  \
0                   3345.814979        6457.753476           3437.0   
1                   3316.354776        6407.964590           3005.0   
2                   3291.158847        6358.027600           2663.0   
3                   3317.891662        6370.000383           2365.0   
4                   3352.846566        6398.171770           2264.0   
...                         ...                ...           

In [75]:
ddd = df_c.copy()

In [72]:
ddd.meningitis = ddd.meningitis/ddd.PopTotal

In [76]:
for i in ddd[cc]:
    ddd[i] = ddd[i]/ddd.PopTotal

In [77]:
ddd.head()

country code  year number_of_executions  meningitis  neoplasms  \
0  Afghanistan  AFG  2007                   15    0.108210   0.587619   
1  Afghanistan  AFG  2008                   17    0.098507   0.582486   
2  Afghanistan  AFG  2009                    0    0.086642   0.576956   
3  Afghanistan  AFG  2011                    2    0.077257   0.567563   
4  Afghanistan  AFG  2012                   14    0.072349   0.562296   

   fire_heat_hot_substances   malaria   drowing  interpersonal_violence  ...  \
0                  0.017757  0.014499  0.078478                0.134950  ...   
1                  0.016672  0.009188  0.071184                0.136522  ...   
2                  0.015788  0.008418  0.065229                0.136437  ...   
3                  0.014871  0.012951  0.058935                0.138454  ...   
4                  0.014288  0.003013  0.055064                0.136216  ...   

   chronic_kidney_disease  chronic_respiratory_deasease  \
0                0.165686                      0.266481   
1                0.163544                      0.257655   
2                0.161895                      0.248118   
3                0.158866                      0.229620   
4                0.155502                      0.220706   

   cirrohosis_liver_deasease  digestive_disease  acute_hepatitis  \
0                   0.123459           0.238289         0.126824   
1                   0.119628           0.231149         0.108397   
2                   0.115907           0.223915         0.093785   
3                   0.110165           0.211506         0.078526   
4                   0.107596           0.205324         0.072654   

   alzhimers_disease  parkinsons_disease   PopTotal  gdp-per-capita  continent  
0           0.051733            0.016605  27100.542        0.053904       Asia  
1           0.051367            0.016413  27722.281        0.053535       Asia  
2           0.051030            0.016200  28394.806        0.061945       Asia  
3           0.050071            0.015705  30117.411        0.063238       Asia  
4           0.049549            0.015468  31161.378        0.066605       Asia  

[5 rows x 39 columns]

In [78]:
ddd.to_csv('datasets/cluster_scraped.csv', index = False)

In [89]:
canc = pd.read_csv('cancer_type.csv')

In [91]:
canc.columns

Index(['country', 'year', 'liver_cancer', 'kidney_cancer',
       'lip_and_oral_cavity_cancer', 'lung', 'larynx', 'gallbladder_biliary',
       'malignant_skin', 'lukemia', 'hodgkin_lymphoma', 'multiple_myeloma',
       'other', 'breast', 'prostate', 'thyroid', 'stomach', 'bladder',
       'uterine', 'ovarian', 'cervical', 'brain', 'non_hodgkin_lymphoma',
       'pancreatic', 'esophageal', 'testicular', 'nasopharynx',
       'other_pharynx', 'colon', 'non_melanoma_skin', 'mesothelioma'],
      dtype='object')